testutil/bech32_codec.go

```diff
@@ -0,0 +1,48 @@
+package testutil
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/cosmos/cosmos-sdk/types/bech32"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type bech32Codec struct {
```

Comment:
When would a Bech32Codec implementation ever be different than this? I can't imagine any case where it'd be different. That being the case, why define it here and not instead in some public package for all to use? Just change `NewBech32Codec()` -> `NewBech32Codec(hrp string)`

Reply:
sure i can make that change, i was thinking making it specific to us but public makes sense

Reply:
Yeah It would be helpful to define something for 99% of use-cases for ppl to use out of the box.

Reply:
added